contracts/service-registry/src/state.rs

```diff
@@ -145,12 +145,46 @@ pub enum AuthorizationState {
     Jailed,
 }
 
-type ServiceName = str;
+pub struct VerifierPerChainIndexes<'a> {
+    pub verifier_address: MultiIndex<
+        'a,
+        (ServiceName, VerifierAddress),
+        (),
+        (ServiceName, ChainName, VerifierAddress),
+    >,
+}
+
+impl<'a> IndexList<()> for VerifierPerChainIndexes<'a> {
+    fn get_indexes(&'_ self) -> Box<dyn Iterator<Item = &'_ dyn Index<()>> + '_> {
+        let v: Vec<&dyn Index<()>> = vec![&self.verifier_address];
+        Box::new(v.into_iter())
+    }
+}
```

Comment:
didn't you use a macro before for this?

Reply:
apparently, that macro does not work when applied to empty value (the updated version of indexed map does not have a value). When added with empty value, build fails saying we need to implement IndexList